ui/components/component-library/text-field/text-field.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { SIZES } from '../../../helpers/constants/design-system';
+
+import { Icon, ICON_NAMES } from '../icon';
+
+import { TextFieldBase } from '../text-field-base';
+
+export const TextField = ({
+  className,
+  showClear,
+  clearIconProps,
+  clearButtonProps,
+  rightAccessory,
+  value: valueProp,
+  onChange,
+  onClear,
+  inputProps,
+  ...props
+}) => {
+  const [value, setValue] = useState(valueProp || '');
+  const handleOnChange = (e) => {
+    setValue(e.target.value);
+    onChange && onChange(e);
```

Comment:
I think you can do `onChange?.(e)`

Reply:
Updated here: https://github.com/MetaMask/metamask-extension/pull/16105/commits/f09595e0bd7d031531ce9d4395a6a14bdac522c6
Fixing up my sloppy code here: https://github.com/MetaMask/metamask-extension/pull/16105/commits/cb5e62dedc2142a6658feaa2f4a57096aa0d8225